crates/circuit/src/vf2.rs

```diff
@@ -0,0 +1,1340 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+// TODO: delete once we move to a version of Rustworkx which includes
```

Comment:
We probably should mention this is forked from rustworkx here right? It's implied by this TODO but not explicitly called out that is copied from rustworkx. I realize it wasn't there before but we should correct this:

```suggestion
// This module was forked from rustworkx at:
// https://github.com/Qiskit/rustworkx/blob/9f0646e8886cfecc55e59b96532c6f7f798524c0/src/isomorphism/vf2.rs

// TODO: delete once we move to a version of Rustworkx which includes
```

Reply:
I rewrote the provenance notice more completely in 1ffadaef93 to keep the full history.  Later in the chain, we should probably put in a note that the file becomes entirely different from Rustworkx's implementation.